src/ram-lib-mapping.ts

```diff
@@ -0,0 +1,520 @@
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  const auctionBasePrice = projectMinterConfig.basePrice;
+
+  if (!auctionBasePrice) {
+    log.warning(
+      "Auction base price for minter {} not found for core contract {}",
+      [event.address.toString(), event.params.coreContract.toHexString()]
+    );
+    return;
+  }
+  const bidValue = slotIndexToBidValue(
+    auctionBasePrice.toString(),
+    event.params.slotIndex
+  );
+
+  const fullProjectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+
+  const ramBidId = generateRAMMinterBidId(
+    event.address.toHexString(),
+    fullProjectId,
+    event.params.bidId.toString()
+  );
+  // @dev: a bid with this ID should not already exist for the RAM Minter
+  let bid = Bid.load(ramBidId);
+  if (bid) {
+    log.warning("Bid ID {} already exists for minter {}", [
+      ramBidId,
+      event.address.toHexString()
+    ]);
+    return;
+  }
+  bid = new Bid(ramBidId);
+  // Create new account entity if one for the bidder doesn't exist
+  const bidderAccount = new Account(event.params.bidder.toHexString());
+  bidderAccount.save();
+
+  bid.project = minterProjectAndConfig.project.id;
+  bid.bidType = "RAM";
+  bid.winningBid = false;
+  bid.minter = event.address.toHexString();
+  bid.value = bidValue;
+  bid.bidder = bidderAccount.id;
+  bid.settled = false;
+  bid.slotIndex = event.params.slotIndex;
+  bid.timestamp = event.block.timestamp;
+  bid.updatedAt = event.block.timestamp;
+  bid.save();
+}
+
+export function handleBidRemoved(event: BidRemoved): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const fullProjectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+
+  const ramBidId = generateRAMMinterBidId(
+    event.address.toHexString(),
+    fullProjectId,
+    event.params.bidId.toString()
+  );
+
+  // @dev: a bid with this ID should exist for the RAM Minter
+  const bid = Bid.load(ramBidId);
+
+  if (!bid) {
+    log.warning("Bid ID {} not found for minter {}", [
+      ramBidId,
+      event.address.toHexString()
+    ]);
+    return;
+  }
+  // Update slotIndex to null, value to 0
+  bid.value = BigInt.fromI32(0);
+  bid.slotIndex = null;
+  bid.updatedAt = event.block.timestamp;
+  bid.save();
```

Comment:
While I think we could delete the bid from the store, updating it in a way that indicates it is deleted and syncing that to db makes sense to me.
Instead of zeroing out value and nulling slotIndex, WDYT about having a field `isRemoved` on the entity or something?

Reply:
Agreed! In favor of being explicit here.. I added the isRemoved field to the Bid entity - do you think it makes sense to update the sea mapping logic to set `isRemoved` to true when a user has been outbid?

Reply:
Yeah I like this and good call I think that behavior makes sense when a user is outbid on SEA as well

Reply:
🙏 Updated!